ci/merge_and_update_env.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
```

Comment:
I fail to understand why you don't like setup_env.sh which saves lots of repetition in the github yaml, and this script is fine. It's nice that this scrip is more atomic and does a more specific thing, but seems like a small detail, considering the value that IMO add not having to duplicate the conda config for every job.

In any cas, I don't understand the criteria, but not a problem. I'm fine with this if you've got a strong opinion of this being better.